ONSdigital/response-operations-ui
# Call new sample summary status endpoint

# What and why?

Prior to this PR, loading a sample meant the sample service would hit the database once to load the sample unit, then again to get the count of all the sample units to see if it was done yet.  If it wasn't then it'd not do anything else, if all the sample units had been loaded then it would transition the state of the sample summary to ACTIVE (which let the things that cared know that the sample load was complete).

This new PR changes that.  During the sample load it does the 1 call per unit that the current one does to save it to the database, but now there is a separate http endpoint that needs to be called to check if all the sample units for a sample summary have been loaded, and then change the status. 

This means that response-operations-ui needs to call that endpoint and find it out if it's done by itself.  We only call the endpoint if the collection exercise is still being created (so basically in the CREATED or SCHEDULED states) AND while the sample summary is still in the INIT state.  If both these are true then we need to keep checking if it's active yet.  This check is only done on 2 pages, the collection exercise page and the sample & CI page for that collection exercise.

This sample PR (https://github.com/ONSdigital/rm-sample-service/pull/232) is tied to this one, and they need to go in as a pair.

# How to test?

- Deploy both PRs to your enviornment
- Create a collection exercise and load a large-ish sample. While it's loading the state should stay as INIT and you should see calls in response-ops hitting the new check endpoint.
- Once they're all loaded, you should see the check endpoint transition the state of the sample summary to ACTIVE.
- Once the state is active, response-ops shouldn't hit that endpoint anymore until you delete the sample.
- Also test deleting the sample and adding it again, the calls should resume.
- Set the exercise live.  Everything should be the same and there should be no more calls to this new endpoint.

# Trello
